app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -55,12 +67,17 @@ export default class SchoolDisciplineDashboard extends React.Component {
     });
   }
 
-  filterStudents(students) {
-    const {selectedChart, selectedCategory} = this.state;
+  filterStudents(students, shouldFilterSelectedCategory) {
+    const {selectedChart, selectedCategory, grade, house, counselor} = this.state;
     return students.filter(student => {
-      if (selectedCategory && selectedChart === 'grade' && student.grade !== selectedCategory) return false;
-      if (selectedCategory && selectedChart === 'homeroom_label' && student.homeroom_label !== selectedCategory) return false;
-      if (selectedCategory && !this.filterIncidents(student.discipline_incidents, true).length) return false;
+      if (grade !== ALL && student.grade !== grade) return false;
+      if (house !== ALL && student.house !== house) return false;
+      if (counselor !== ALL && student.counselor !== counselor) return false;
+      if (selectedCategory && shouldFilterSelectedCategory) { //used by the student list when a user selects a category within a chart
+        if (selectedChart === 'grade' && student.grade !== selectedCategory) return false;
```

Comment:
This seems like there are two separate bits of state related to the grade filtering.  One is for if the user uses the select filter, and the other is if they click on a grade within a chart.

Could these be a single bit of state?  In other words, when the user clicks on a grade within a chart, would it simplify to call `this.setState({grade})` or to call `this.setsState({homeroom})`?

Reply:
We could certainly do that. It might be awkward to treat selecting a grade within a chart different from selecting other categories within a chart. We would still have need state for selectedCategory, but have an exception when that category is a grade. Since there are two ways you might filter a grade and the application behavior is different depending on which is selected, it seemed to make sense to hold that as two different bits of state. 

I imagine moving forward that we might not need the grade/homeroom charts with the top level filters, in which case this problem should go away.